src/main/java/http/request/RequestRouter.java

```diff
@@ -7,33 +7,41 @@
 import java.nio.charset.StandardCharsets;
 
 
-import static http.request.RequestParser.*;
-
 public class RequestRouter {
     public static void requestReader(InputStream input) throws IOException {
         BufferedReader reader = new BufferedReader(new InputStreamReader(input));
 
-        setRequestType(reader.readLine());
 
-        Utils.print(Messages.echoReceived(reader.readLine()));
+        Utils.print(Messages.httpReceived(reader.readLine()));
         Utils.print("output read");
-
+        parseRequest(reader.readLine());
     }
 
-    public static void setupResponse() {
-        setHeaders("text/html");
-        setRequestBody("hello".getBytes(StandardCharsets.UTF_8));
-        setStatus("200", "well done :)");
+    public static String[] parseRequest(String requestLine) {
+        String[] splitData = requestLine.split(" ");
+        return splitData;
     }
+ public static Request buildRequest (String method, String path, String protocol) {
+        Request request = new Request();
+        request.setMethod(method);
+        request.setPath(path);
+        request.setHeaders("protocol",protocol);
+        return request;
```

Comment:
Am I correct that your goal here is to:
- use the `parseRequest` and `buildRequest` to transform the inpoutStream into the `request` object (partially coded already)
- then (not coded yet) add to this class a routing function that will analyse the request object to then handle it accordingly?

Reply:
yes that's my plan.